kratos/conditions/geometrical_condition.h

```diff
@@ -0,0 +1,287 @@
+     * @param NewId The Id of the new created condition
+     * @param pGeometry The pointer to the geometry that will define the condition
+     * @param pProperties The pointer to the properties that will define the behaviour of the condition
+     */
+    GeometricalCondition(
+        IndexType NewId, 
+        GeometryType::Pointer pGeometry, 
+        PropertiesType::Pointer pProperties
+        );
+
+    ///Copy constructor
+    GeometricalCondition(GeometricalCondition const& rOther);
+
+    /// Destructor.
+    ~GeometricalCondition() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    GeometricalCondition& operator=(GeometricalCondition const& rOther);
+
+    ///@}
+    ///@name Operations
+    ///@{
+   
+    /**
+     * @brief Creates a new condition pointer
+     * @param NewId the ID of the new condition
+     * @param ThisNodes the nodes of the new condition
+     * @param pProperties the properties assigned to the new condition
+     * @return a Pointer to the new condition
+     */
+    Condition::Pointer Create(
+        IndexType NewId, 
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties
+        ) const override;
+
+    /**
+     * @brief Creates a new condition pointer
+     * @param NewId the ID of the new condition
+     * @param pGeom the geometry to be employed
+     * @param pProperties the properties assigned to the new condition
+     * @return a Pointer to the new condition
+     */
+    Condition::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties
+        ) const override;
+
+    /**
+     * @brief Creates a new condition pointer and clones the previous condition data
+     * @param NewId the ID of the new condition
+     * @param ThisNodes the nodes of the new condition
+     * @param pProperties the properties assigned to the new condition
+     * @return a Pointer to the new condition
+     */
+    Condition::Pointer Clone (
+        IndexType NewId, 
+        NodesArrayType const& ThisNodes
+        ) const override;
+
+    ///@}
+    ///@name Access
+    ///@{
+    ///@}
+    ///@name Inquiry
+    ///@{
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "Geometrical Condition #" << Id();
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "Geometrical Condition #" << Id();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        pGetGeometry()->PrintData(rOStream);
+    }
+    
+    ///@}
+    ///@name Friends
+    ///@{
+    ///@}
+
+protected:
+    
+    ///@name Protected static Member Variables
```

Comment:
I would delete all empty sections here, to make the class a bit more compact and faster to read.